backend/dataall/tasks/data_sharing/share_managers/s3_share_manager.py

```diff
@@ -196,7 +196,11 @@ def manage_access_point_and_policy(self):
             )
             access_point_arn = S3.create_bucket_access_point(self.source_account_id, self.source_environment.region, self.bucket_name, self.access_point_name)
             # Access point creation is slow
-            time.sleep(5)
+            while not S3.get_bucket_access_point_arn(self.source_account_id, self.source_environment.region, self.access_point_name):
+                logger.info(
+                    'Waiting 30s for access point creation to complete..'
+                )
+                time.sleep(30)
         existing_policy = S3.get_access_point_policy(self.source_account_id, self.source_environment.region, self.access_point_name)
```

Comment:
Not to request information twice we could do:
```
    access_point_arn = S3.create_bucket_access_point(self.source_account_id, self.source_environment.region, self.bucket_name, self.access_point_name)
existing_policy = None
while not existing_policy:
        waiting....
        existing_policy = S3.get_access_point_policy....
```

Reply:
The policy should not exist even after we create the access point. So checking the policy existence is not going to solve the issue. The error arises when we try to attach a new policy to a still-creating access point. BUt open to ideas on how to make it better, I coded it a bit fast

Reply:
Thanks for the clarification!